src/py_zfs_pool.c

```diff
@@ -413,7 +283,31 @@ PyTypeObject ZFSPool = {
 	.tp_init = py_zfs_pool_init,
 	.tp_doc = "ZFSPool",
 	.tp_dealloc = (destructor)py_zfs_pool_dealloc,
-	.tp_str = py_zfs_pool_str,
-	.tp_repr = py_zfs_pool_str,
+	.tp_repr = py_repr_zfs_pool,
 	.tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
 };
+
+py_zfs_pool_t *init_zfs_pool(py_zfs_t *lzp, zpool_handle_t *zhp)
+{
+	py_zfs_pool_t *out = NULL;
+
+	out = (py_zfs_pool_t *)PyObject_CallFunction((PyObject *)&ZFSPool, NULL);
+	if (out == NULL) {
+		return (NULL);
+	}
+
+	out->pylibzfsp = lzp;
+	Py_INCREF(lzp);
+
+	Py_BEGIN_ALLOW_THREADS
+	out->name = PyUnicode_FromString(zpool_get_name(zhp));
```

Comment:
We shouldn't create new python objects with GIL released.

Reply:
Updated, thanks.